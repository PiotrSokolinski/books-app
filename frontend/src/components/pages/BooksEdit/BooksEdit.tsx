import { Edit, ShowButton, useTranslate } from 'react-admin';
import { useParams } from 'react-router-dom';

import { BookActions } from '../../organisms/BookActions/BookActions';
import { BookEditForm } from '../../organisms/BookEditForm/BookEditForm';
import { Wrapper } from '../../templates/Wrapper/Wrapper';

export const BooksEdit = () => {
  const translate = useTranslate();
  const { id } = useParams();

  return (
    <Wrapper>
      <Edit
        id={id}
        title={translate('booksEdit.barTitle')}
        actions={<BookActions action={<ShowButton />} />}
      >
        <BookEditForm />
      </Edit>
    </Wrapper>
  );
};
