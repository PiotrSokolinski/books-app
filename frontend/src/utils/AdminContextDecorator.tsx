import React from 'react';
import {
  AdminContext,
  DataProvider,
  GetListResult,
  memoryStore,
  testDataProvider,
} from 'react-admin';

import { theme } from '../theme/theme';
import { i18nProvider } from '../i18n/i18n';
import { Book } from '../types/book.ts';

const defaultDataProvider: DataProvider = testDataProvider({
  getList: async (): Promise<GetListResult<Book>> => {
    const books: Book[] = [
      {
        id: '1',
        title: 'The Hobbit',
        author: 'J.R.R. Tolkien',
        isbnNumber: '1234567890',
        numberOfPages: 300,
        rating: 4,
      },
      {
        id: '2',
        title: 'The Lord of the Rings',
        author: 'J.R.R. Tolkien',
        isbnNumber: '0987654321',
        numberOfPages: 1000,
        rating: 5,
      },
    ];

    return Promise.resolve({
      total: books.length,
      data: books,
    });
  },
  getOne: () => {
    return Promise.resolve({
      data: {
        id: '1',
        title: 'The Lord of the Rings',
        author: 'J.R.R. Tolkien',
        isbnNumber: '0987654321',
        numberOfPages: 1000,
        rating: 5,
      },
    });
  },
  create: () => {
    return Promise.resolve({
      data: {
        id: '3',
        title: 'Test Book Title',
        author: 'J.R.R. Tolkien',
        isbnNumber: '0987654321',
        numberOfPages: 1000,
        rating: 5,
      },
    });
  },
  update: () => {
    return Promise.resolve({
      data: {
        id: '2',
        title: 'Test Book Title',
        author: 'J.R.R. Tolkien',
        isbnNumber: '0987654321',
        numberOfPages: 1000,
        rating: 5,
      },
    });
  },
  delete: () => {
    return Promise.resolve({
      data: {
        id: '3',
        title: 'Test Book Title',
        author: 'J.R.R. Tolkien',
        isbnNumber: '0987654321',
        numberOfPages: 1000,
        rating: 5,
      },
    });
  },
} as unknown as DataProvider); // https://github.com/marmelab/react-admin/issues/5476

// TODO: F7 - Add more documentation and comments to the codebase
/**
 * A decorator component that provides the React-Admin context for testing or storybook environments.
 *
 * This component wraps its children with the necessary React-Admin context providers,
 * including a theme, internationalization provider, and optionally a `dataProvider`.
 * The `authProvider` is set to `undefined`, as this provider is designed for simple use cases.
 *
 * It supports passing a custom `dataProvider` for testing or storybook purposes.
 *
 * @param {Object} props - The props for the component.
 * @param {React.ReactNode} props.children - The child components to render within the React-Admin context.
 * @param {Partial<DataProvider>} [props.dataProvider] - An optional custom `dataProvider` to be used in place of the default one.
 *
 * @returns {JSX.Element} The wrapped children within the React-Admin context, with context providers like theme and internationalization applied.
 *
 * @example
 * <AdminContextDecorator>
 *   <MyComponent />
 * </AdminContextDecorator>
 */
export const AdminContextDecorator: React.FC<{
  children: React.ReactNode;
  dataProvider?: Partial<DataProvider>;
}> = ({
  children,
  dataProvider,
}: {
  children: React.ReactNode;
  dataProvider?: Partial<DataProvider>;
}): React.JSX.Element => (
  <AdminContext
    dataProvider={
      dataProvider ? testDataProvider(dataProvider) : defaultDataProvider
    }
    authProvider={undefined}
    theme={theme}
    store={memoryStore()}
    i18nProvider={i18nProvider}
  >
    {children}
  </AdminContext>
);
