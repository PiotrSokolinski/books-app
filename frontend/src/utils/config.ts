const { VITE_API_URL } = import.meta.env;

export const APP_CONFIG = {
  VITE_API_URL,
};
