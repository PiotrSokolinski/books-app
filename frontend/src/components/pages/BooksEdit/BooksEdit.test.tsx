import { render, screen, waitFor } from '@testing-library/react';
import { vi, describe, it, expect } from 'vitest';
import { fireEvent } from '@testing-library/react';
import { ResourceContextProvider } from 'react-admin';

import { BooksEdit } from './BooksEdit';
import { AdminContextDecorator } from '../../../utils/AdminContextDecorator';

vi.mock(import('react-admin'), async (importOriginal) => {
  const actual = await importOriginal();

  return {
    ...actual,
    useRedirect: vi.fn(() => vi.fn()),
    useEditController: vi.fn().mockReturnValue({
      record: {
        id: '1',
        title: 'Test Book',
        author: 'Test Author',
        isbnNumber: '1234567890',
        numberOfPages: 200,
        rating: 5,
      },
    }),
  };
});

vi.mock(import('react-router-dom'), async (importOriginal) => {
  const actual = await importOriginal();

  return {
    ...actual,
    useParams: vi.fn().mockReturnValue({ id: '1' }),
  };
});

vi.mock('react-hook-form', async () => {
  const actual = await import('react-hook-form');

  return {
    ...actual,
    useFormContext: vi.fn().mockReturnValue({
      reset: vi.fn(),
    }),
  };
});

describe('BooksEdit', () => {
  it('renders the BooksEdit component and submits a record', async () => {
    render(
      <AdminContextDecorator>
        <ResourceContextProvider value="books">
          <BooksEdit />
        </ResourceContextProvider>
      </AdminContextDecorator>,
    );

    await waitFor(() => {
      fireEvent.change(screen.getByLabelText(/title/i), {
        target: { value: 'Edited Book Title' },
      });
    });

    const saveButton = screen.getByText(/save/i);
    expect(saveButton).toBeInTheDocument();
    fireEvent.click(saveButton);
  });

  it('renders the BooksEdit component and deletes a record', async () => {
    render(
      <AdminContextDecorator>
        <ResourceContextProvider value="books">
          <BooksEdit />
        </ResourceContextProvider>
      </AdminContextDecorator>,
    );

    await waitFor(() => {
      const deleteButton = screen.getByText(/delete/i);
      expect(deleteButton).toBeInTheDocument();
      fireEvent.click(deleteButton);
    });

    const confirmButton = screen.getByText(/confirm/i);
    expect(confirmButton).toBeInTheDocument();
    fireEvent.click(confirmButton);
  });

  it('matches the snapshot', async () => {
    const { asFragment } = render(
      <AdminContextDecorator>
        <ResourceContextProvider value="books">
          <BooksEdit />
        </ResourceContextProvider>
      </AdminContextDecorator>,
    );

    await waitFor(() => {
      const confirmButton = screen.getByText(/delete/i);
      expect(confirmButton).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
