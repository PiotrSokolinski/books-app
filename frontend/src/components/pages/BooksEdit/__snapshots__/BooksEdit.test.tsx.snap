// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BooksEdit > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1es8s0n"
  >
    <div
      class="edit-page css-158ezdf-RaEdit-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-dense css-8czsxw-MuiToolbar-root-RaTopToolbar-root"
      >
        <div
          class="MuiBox-root css-1oxffa5"
        >
          <button
            aria-label="Back to list"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-7nabzj-MuiButtonBase-root-MuiButton-root-RaButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"
                />
              </svg>
            </span>
            Back to list
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <a
            aria-label="Show"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5g6lod-MuiButtonBase-root-MuiButton-root-RaButton-root"
            href="#/books/1/show"
            tabindex="0"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="RemoveRedEyeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
                />
              </svg>
            </span>
            Show
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
      <div
        class="RaEdit-main"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root RaEdit-card css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <form>
            <div
              class="MuiCardContent-root css-1fd8x6x-MuiCardContent-root-RaSimpleForm-root"
            >
              <div
                class="MuiStack-root css-1a69w1n-MuiStack-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5 css-1iranwn-MuiTypography-root"
                >
                  Edit The Lord of the Rings
                </h5>
                <div
                  class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root ra-input ra-input-title css-1i5wpaj-MuiFormControl-root-MuiTextField-root-RaResettableTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled css-11qvwfa-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for=":rm:"
                    id=":rm:-label"
                  >
                    <span>
                      Title
                      <span
                        aria-hidden="true"
                      >
                         *
                      </span>
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1lm524j-MuiInputBase-root-MuiFilledInput-root"
                  >
                    <input
                      aria-describedby=":rm:-helper-text"
                      aria-invalid="false"
                      class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-1zuho2-MuiInputBase-input-MuiFilledInput-input"
                      id=":rm:"
                      name="title"
                      type="text"
                      value="The Lord of the Rings"
                    />
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained MuiFormHelperText-filled css-k4qjio-MuiFormHelperText-root"
                    id=":rm:-helper-text"
                  >
                    <span>
                      ​
                    </span>
                  </p>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root ra-input ra-input-author css-1i5wpaj-MuiFormControl-root-MuiTextField-root-RaResettableTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled css-11qvwfa-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for=":ro:"
                    id=":ro:-label"
                  >
                    <span>
                      Author
                      <span
                        aria-hidden="true"
                      >
                         *
                      </span>
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1lm524j-MuiInputBase-root-MuiFilledInput-root"
                  >
                    <input
                      aria-describedby=":ro:-helper-text"
                      aria-invalid="false"
                      class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-1zuho2-MuiInputBase-input-MuiFilledInput-input"
                      id=":ro:"
                      name="author"
                      type="text"
                      value="J.R.R. Tolkien"
                    />
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained MuiFormHelperText-filled css-k4qjio-MuiFormHelperText-root"
                    id=":ro:-helper-text"
                  >
                    <span>
                      ​
                    </span>
                  </p>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root ra-input ra-input-isbnNumber css-1i5wpaj-MuiFormControl-root-MuiTextField-root-RaResettableTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled css-11qvwfa-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for=":rq:"
                    id=":rq:-label"
                  >
                    <span>
                      ISBN number
                      <span
                        aria-hidden="true"
                      >
                         *
                      </span>
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1lm524j-MuiInputBase-root-MuiFilledInput-root"
                  >
                    <input
                      aria-describedby=":rq:-helper-text"
                      aria-invalid="false"
                      class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-1zuho2-MuiInputBase-input-MuiFilledInput-input"
                      id=":rq:"
                      name="isbnNumber"
                      type="text"
                      value="0987654321"
                    />
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained MuiFormHelperText-filled css-k4qjio-MuiFormHelperText-root"
                    id=":rq:-helper-text"
                  >
                    <span>
                      ​
                    </span>
                  </p>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root ra-input ra-input-numberOfPages css-m4dhox-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled css-11qvwfa-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for=":rs:"
                    id=":rs:-label"
                  >
                    <span>
                      Number of pages
                      <span
                        aria-hidden="true"
                      >
                         *
                      </span>
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1lm524j-MuiInputBase-root-MuiFilledInput-root"
                  >
                    <input
                      aria-describedby=":rs:-helper-text"
                      aria-invalid="false"
                      class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-1zuho2-MuiInputBase-input-MuiFilledInput-input"
                      id=":rs:"
                      name="numberOfPages"
                      step="any"
                      type="number"
                      value="1000"
                    />
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained MuiFormHelperText-filled css-k4qjio-MuiFormHelperText-root"
                    id=":rs:-helper-text"
                  >
                    <span>
                      ​
                    </span>
                  </p>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root ra-input ra-input-rating css-m4dhox-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled css-11qvwfa-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for=":ru:"
                    id=":ru:-label"
                  >
                    <span>
                      Rating
                      <span
                        aria-hidden="true"
                      >
                         *
                      </span>
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1lm524j-MuiInputBase-root-MuiFilledInput-root"
                  >
                    <input
                      aria-describedby=":ru:-helper-text"
                      aria-invalid="false"
                      class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-1zuho2-MuiInputBase-input-MuiFilledInput-input"
                      id=":ru:"
                      name="rating"
                      step="any"
                      type="number"
                      value="5"
                    />
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained MuiFormHelperText-filled css-k4qjio-MuiFormHelperText-root"
                    id=":ru:-helper-text"
                  >
                    <span>
                      ​
                    </span>
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1iw8x7s-MuiToolbar-root"
            >
              <button
                aria-label="Save"
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1f0c020-MuiButtonBase-root-MuiButton-root-RaSaveButton-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SaveIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3m3-10H5V5h10z"
                  />
                </svg>
                Save
              </button>
              <button
                aria-label="Delete"
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-r920ud-MuiButtonBase-root-MuiButton-root-RaButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DeleteIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                    />
                  </svg>
                </span>
                Delete
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
